web/assets/js/actions.ts

```diff
@@ -0,0 +1,125 @@
+import { io } from "socket.io-client";
+import { saveToLocalStorage, getFromLocalStorage } from "./utils/localstorage";
+
+export default new (class {
+  constructor() {
+    //INFO: MVP -> May not be the best way to handle the gno-ts
```

Comment:
This is actually the best way to go. The gno-js library is (intentionally) made with exclusive reliance on the JSON-RPC server of the node, meaning there is no state management locally (apart from something that's temporarily stored in memory) that needs to happen for transactions to be processed by the node. 

The `Provider` and `Wallet` instances can be saved in this class, as they are reused for all the functionality, and are available throughout the dApp. The constructor obviously just needs to instantiate them again, to prepare them for usage (connect to the JSON-RPC server of the node, init the Wallet)

Reply:
Cool that's what I thought after reading your readme document.